src/app/game-state/home.service.ts

```diff
@@ -698,35 +699,52 @@ export class HomeService {
   }
 
   autoBuy(){
+    let priceBuffer = (this.home.costPerDay + 1) * 14; // Two weeks, by popular request.
     if (this.autoBuyHomeUnlocked && this.homeValue < this.autoBuyHomeLimit){
+      // check for final building to avoid buying excess building land.
+      if (!this.upgrading && this.nexthome.HomeType < this.autoBuyHomeLimit){
```

Comment:
Why the extra check on the home limit? Does the if clause above not already do that?

Reply:
Because this is for the next home, not the current home, and I wasn't sure if you could just +1 the value. If I got it right, the logic *should* be that you only buy land if you need it to build the next home, or that you're not already building the final home. Upon review I think the not needs to be outside a parenthetical for this to make sense, there's so much logic going on in here I can't be sure.

Current behavior is that while you're upgrading the final home in the buy limit, you buy land for it even though you don't need it. A tiny issue unless you decide to change the price scaling of land in the future. Honestly I'd suggest doing away with buying land and make it purely fields and a price for home since empty lots aren't used for anything I've seen, unless I'm missing something in impossible tasks.

Reply:
And now I realize that check isn't needed because if you're not upgrading you can't get past that first check if you're done. I blame the sleep deprivation.